### PR TITLE
SWARM-858: additional fractions: unintuitive behavior if the fraction is not found

### PR DESCRIPTION
**Pull request by Ladicek**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [ ] Have you built the project locally prior to submission with `mvn clean install`?

-----

Motivation
----------
If the user sets an additional fraction in the Maven plugin
configuration but makes a mistake, two things can happen:

1. If the user used the format "artifactId", an unhelpful
   error message is shown.
2. If the user used the format "artifactId:version"
   or "groupId:artifactId:version", build passes just fine.

Both of these behaviors are unintuitive.

Modifications
-------------
The method FractionDescriptor.fromGav was changed to always
consult the FractionList. If the fraction isn't found
in the list, an appropriate error message is used.

Result
------
Wrong configuration of additional fractions will now always
result in build errors.

### CODE REVIEW

File: tools/src/main/java/org/wildfly/swarm/tools/FractionDescriptor.java [hunk 1]
```diff
@@ -39,24 +39,50 @@ public FractionDescriptor(String groupId, String artifactId, String version) {
...
+                    throw new RuntimeException("Version mismatch: requested " + gav + ", found " + desc.av());
+                }
                 break;
             case 3:
-                desc = new FractionDescriptor(parts[0], parts[1], parts[2]);
+                desc = fractionList.getFractionDescriptor(parts[0], parts[1]);
+                if (desc == null) {
+                    throw new RuntimeException("Fraction not found: " + gav);
+                }
+                if (!desc.getVersion().equals(parts[1])) {
```
Comment (kenfinnigan) [inline tools/src/main/java/org/wildfly/swarm/tools/FractionDescriptor.java]: Should this be parts[2] ??

Comment (Ladicek) [reply tools/src/main/java/org/wildfly/swarm/tools/FractionDescriptor.java]: Oops! Absolutely.

Comment (Ladicek) [reply tools/src/main/java/org/wildfly/swarm/tools/FractionDescriptor.java]: Fixed, thanks.

**Review by kenfinnigan — APPROVED** [review]

**Comment by Ladicek** [comment]
Note to self: I need to check if the old code didn't support some sort of wildcards, especially for the version field. That would be regressed by this commit.

**Pull request merged**